### DISCUSSION TITLE
Foreign Key Constraint - Create Record

### DISCUSSION THREAD

**Discussion by josephbobby24** [prisma/prisma#15725] [Q&A] [answered]
### Bug description

Trying to create a record of Model "Commission"

Code: 
```ts
        this.client.db.commission.create({
            data: {
                channelId: "a",
                department: "a",
                details: ["Detail1"],
                roleId: "b",
                clientUserId: "a",
            }
        }).then(comm => {
            console.log(comm)
        })`
```
```
Error:
private async loadTickets() {
→ 12     this.client.db.commission.create(
Foreign key constraint failed on the field: `clientUserId`
    at RequestHandler.handleRequestError (C:\Users\josep\OneDrive\Desktop\Bots\Navi\node_modules\@prisma\client\runtime\index.js:29909:13)
    at RequestHandler.request (C:\Users\josep\OneDrive\Desktop\Bots\Navi\node_modules\@prisma\client\runtime\index.js:29892:12)
    at async PrismaClient._request (C:\Users\josep\OneDrive\Desktop\Bots\Navi\node_modules\@prisma\client\runtime\index.js:30864:16) {
  code: 'P2003',
  clientVersion: '4.3.1',
  meta: { field_name: 'clientUserId' }
}
```
### How to reproduce

Run the code linked above, attempting to create a record that has a relational field.

### Expected behavior

A record is created

### Prisma information
```prisma
datasource db {
  provider = "mysql"
  url      = ""
}

model Commission {
  id           Int         @unique @default(autoincrement())
  clientUserId       String
  channelId    String
  department   String
  roleId       String
  details      Json
  freelancerId String?
  managerId    String?
  Freelancer   Freelancer? @relation(fields: [freelancerId], references: [userId])
  Manager      Manager?    @relation(fields: [managerId], references: [userId])
  Client       Client?      @relation(fields: [clientUserId], references: [userId])
}

model Profile {
  userId     String      @unique
  Freelancer Freelancer?
  Client     Client?
  Manager    Manager?
}

model Manager {
  userId      String       @unique
  Profile     Profile?     @relation(fields: [userId], references: [userId])
  Commissions Commission[]
}

model Freelancer {
  userId      String       @unique
  portfolio   String       @default("No portfolio!")
  timezone    String?
  paypal      String       @default("No paypal set!")
  Profile     Profile?     @relation(fields: [userId], references: [userId])
  Commissions Commission[]
}

model Client {
  userId      String       @unique
  Profile     Profile?     @relation(fields: [userId], references: [userId])
  Commissions Commission[]
}
```

### Environment & setup

Windows 11, Node 16

### Prisma Version

prisma                  : 4.3.1
@prisma/client          : 4.3.1
Current platform        : windows

**Comment by Jolg42** [comment]
Did you check that the `Client` exists? Maybe this errored because you tried to reference a non existing `Client` with `clientUserId`

**Comment by josephbobby24** [comment]
It doesn’t, but should it not make one?

**Comment by Jolg42** [comment]
No not with that query,
You can use the nested API with `connectOrCreate` to either connect if already existing or create it if not

```ts
    await prisma.customer.create({
      data: {
        channelId: "a",
        department: "a",
        details: ["Detail1"],
        roleId: "b",
        client: {
          connectOrCreate: {
            create: {
              userId: "a",
            },
            where: {
              userId: "a",
            },
          },
        },
      },
    });
```


See https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connectorcreate

**Comment by josephbobby24** [comment]
I believe I tried that already, but I’ll try it again when I am home.
Thank you, will let you know.

**Comment by josephbobby24** [comment]
Does not work, @Jolg42 

Code: 
`this.client.db.commission.create({
            //@ts-ignore
            data: {
                channelId: "a",
                department: "a",
                details: "a",
                roleId: "a",
                clientUserId: "a",
                Client: {
                    connectOrCreate: {
                        create: {
                            userId: "a" as unknown as never,
                            Profile: {
                                connectOrCreate: {
                                    create: {userId: "a" as unknown as never},
                                    where: {userId: "a" as unknown as never}
                                }
                            }
                        },
                        where: {
                            userId: "a"
                        }
                    }
                }
            }
        }).then(record => {
            console.log(record)
        })`

Error:
Unknown arg `Client` in data.Client for type CommissionUncheckedCreateInput. Did you mean `select`? Available args:

**Reply by josephbobby24** [reply]
Maybe it’s because of multiple relational fields? Schema is above so let me know

**Comment by antja03** [comment, marked as answer]
This error was caused by including both `clientUserId` and `Client` args in the query. Since `clientUserId` is the relation field for `Client`, the `create` function was expecting UncheckedCreateInput which wouldn't include properties for relation tables i.e `Client`

Edit: Just to clarify this is describing the error
> Error:
Unknown arg Client in data.Client for type CommissionUncheckedCreateInput. Did you mean select? Available args:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
